### ISSUE TITLE
add option to change unitsPerEm of compiled font?

### ISSUE THREAD

**Issue by anthrotype** [googlefonts/ufo2ft#376]
it turns out one can [use fontMath to change the UPEM of a font](https://gist.github.com/anthrotype/62d0bfe1d38b8f11a199bd3b66574bcc).
This could be useful if one wishes to increase (or even decrease) the resolution of the font design units.
E.g. merging one font with another with different unitsPerEm, or designing with a given unitsPerEm but then scale everything up to reduce rounding errors when generating intermediate instances.

I was thinking maybe such thing could become a ufo2ft preProcessor filter. And the compile functions could have an `unitsPerEm=None` option that, when set, would trigger this filter.

It would compare the requested unitsPerEm with the UFO's current unitsPerEm and scale the output font accordingly to fit the desired UPEM.

Is this too niche or would this be something desirable to have?

**Comment by madig** [comment]
@davelab6 didn't you think about increasing upMs for VFs generally for smoother interpolation?

**Comment by khaledhosny** [comment]
Such an option will need also to scale GPOS data that are stored in the feature file.

**Comment by anthrotype** [comment]
you're right. We also need to scale all the numbers expressed in font design units that occur in GPOS value records or anchor statements and others. 
Basically search for keyword `<metric>` in http://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html

**Comment by anthrotype** [comment]
By the way, fontMath seens to be doing more than just scaling: it's filtering points that it deems unnecessary, converting lines into curves, etc. which means scaled masters are not guaranteed to keep interpolation compatibility. So I changed that scale_ufo.py script of mine to use ufo2ft's TransformationsFilter to scale the glyphs, and multiply the fontinfo and kerning directly without the aid of fontMath.
